include/beast/websocket/detail/stream_base.hpp

```diff
@@ -321,79 +396,96 @@ read_fh2(DynamicBuffer& db, close_code::value& code)
     case 127:
     {
         std::uint8_t b[8];
-        assert(buffer_size(db.data()) >= sizeof(b));
+        BOOST_ASSERT(buffer_size(db.data()) >= sizeof(b));
         db.consume(buffer_copy(buffer(b), db.data()));
-        rd_fh_.len = big_uint64_to_native(&b[0]);
+        fh.len = big_uint64_to_native(&b[0]);
         // length not canonical
-        if(rd_fh_.len < 65536)
+        if(fh.len < 65536)
         {
             code = close_code::protocol_error;
             return;
         }
         break;
     }
     }
-    if(rd_fh_.mask)
+    if(fh.mask)
     {
         std::uint8_t b[4];
-        assert(buffer_size(db.data()) >= sizeof(b));
+        BOOST_ASSERT(buffer_size(db.data()) >= sizeof(b));
         db.consume(buffer_copy(buffer(b), db.data()));
-        rd_fh_.key = little_uint32_to_native(&b[0]);
+        fh.key = little_uint32_to_native(&b[0]);
     }
     else
     {
         // initialize this otherwise operator== breaks
-        rd_fh_.key = 0;
+        fh.key = 0;
     }
-    if(rd_fh_.mask)
-        prepare_key(rd_key_, rd_fh_.key);
-    if(! is_control(rd_fh_.op))
+    if(! is_control(fh.op))
     {
-        if(rd_fh_.op != opcode::cont)
+        if(fh.op != opcode::cont)
         {
-            rd_size_ = rd_fh_.len;
-            rd_opcode_ = rd_fh_.op;
+            rd_.size = 0;
+            rd_.op = fh.op;
         }
         else
         {
-            if(rd_size_ > (std::numeric_limits<
-                std::uint64_t>::max)() - rd_fh_.len)
+            if(rd_.size > (std::numeric_limits<
+                std::uint64_t>::max)() - fh.len)
             {
                 code = close_code::too_big;
                 return;
             }
-            rd_size_ += rd_fh_.len;
+            //rd_.size += fh.len;
         }
-        if(rd_msg_max_ && rd_size_ > rd_msg_max_)
+#if 0
+        if(rd_msg_max_ && rd_.size > rd_msg_max_)
         {
             code = close_code::too_big;
             return;
         }
-        rd_need_ = rd_fh_.len;
-        rd_cont_ = ! rd_fh_.fin;
+#endif
```

Comment:
No, quite the opposite we should generate a compiler warning, this is lost functionality that we need to soonish figure out how to get back.